# HG changeset patch
# User jwe
# Date 1201030230 0
#      Tue Jan 22 19:30:30 2008 +0000
# Branch release-3-0-x
# Node ID 3257cccb9ed79a3471a8f841ba6ed0c6a3b0951a
# Parent  ca0cbc46abce4fe5dea13e0f1ca6541679aad500
[3-0-0-branch @ 2008-01-22 19:30:30 by jwe]

diff --git a/liboctave/ChangeLog b/liboctave/ChangeLog
--- a/liboctave/ChangeLog
+++ b/liboctave/ChangeLog
@@ -1,8 +1,13 @@
+2008-01-22  Michael Goffioul <michael.goffioul@gmail.com>
+
+	* oct-time.cc (octave_base_tim::init): Validate pointer argument;
+	this fixes the "localtime(-1)" crash under Windows.
+
 2008-01-07  David Bateman  <dbateman@free.fr>
 
 	* Sparse-op-defs.h (SPARSE_ANY_ALL_OP_ROW_CODE): Don't break from
 	loop if this test succeeds.
 
 2008-01-03  David Bateman  <dbateman@free.fr>
 
 	* MSparse.cc (SPARSE_A2A2_OP): If first arg is scalar zero, then
diff --git a/liboctave/oct-time.cc b/liboctave/oct-time.cc
--- a/liboctave/oct-time.cc
+++ b/liboctave/oct-time.cc
@@ -286,16 +286,19 @@ octave_base_tm::strftime (const std::str
     }
 
   return retval;
 }
 
 void
 octave_base_tm::init (void *p)
 {
+  if (! p)
+    return;
+
   struct tm *t = static_cast<struct tm*> (p);
   
   tm_sec = t->tm_sec;
   tm_min = t->tm_min;
   tm_hour = t->tm_hour;
   tm_mday = t->tm_mday;
   tm_mon = t->tm_mon;
   tm_year = t->tm_year;
